systemd_ctypes/bus.py

```diff
@@ -125,6 +125,28 @@ def get_body(self):
         self.rewind(True)
         return tuple(self.yield_values())
 
+    def send(self):
+        self.get_bus().send(self, None)
+        return True
+
+    def reply_method_error(self, error):
+        return self.new_method_error(error).send()
+
+    def reply_method_return(self, out_types, result):
+        # In the general case, a function returns an n-tuple, but
+        # we special-case n=0 and n=1 to be more human-friendly.
+        # For n=0, we expect None.  For n=1 we just take the value.
+        if len(out_types) == 0:
+            assert result is None
+            result = ()
+        elif len(out_types) == 1:
+            result = (result,)
```

Comment:
TBH I don't like this much. While it might be convenient, it's brittle/unexpected and breaks e.g. passing on some result with `*args`. I'd prefer to always pass a tuple, and asserting that.

As a compromise, this method could take `*result`? This would avoid the special-case, and you still retain the convenience.

Reply:
The issue is that we call this function with the return value of the handler function.

Returning one-tuples for functions that return a single value would be deeply strange.

Reply:
Maybe, but with `*result` it would look normal, and returning more than one value would look nice and consistent as well IMHO.